### ISSUE TITLE
Provide Kubernetes version compatibility matrix

### ISSUE THREAD

**Issue by diversario** [argoproj/argo-events#2979]
**Is your feature request related to a problem? Please describe.**
I am preparing for a round of cluster upgrades and I'm trying to understand what Kubernetes versions (1.27+) Argo Events supports.

**Describe the solution you'd like**
A table like the [one provided by Argo Workflows](https://argo-workflows.readthedocs.io/en/latest/releases/#kubernetes-compatibility-matrix), clearly stating AE vs k8s version compatibility.

**Describe alternatives you've considered**

**Additional context**
This feature request is identical to https://github.com/argoproj/argo-workflows/issues/8198 and https://github.com/argoproj/argo-cd/issues/12276.

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by mrharpo** [comment]
+1 for a `k8s` compatibility table. We have deferred upgrading our cluster for this reason, and are starting to incur costs due to the lack of upgrades, so I started digging to find an answer.

Please correct me if anything is incorrect:

## Current supported version
As I understand, [the current `k8s.io/api` version is `0.29.2`](https://github.com/argoproj/argo-events/blob/473ecd7c868d1509121532da8af0cc1face70a12/go.mod#L88), updated [8 months ago](https://github.com/argoproj/argo-events/commit/8ece8fd641b93034b347114b2050ee9d22787f93#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L86-R88) in #3107


Does this mean the latest currently supported Kubernetes version is `1.29.2` ?

**Comment by tooptoop4** [comment]
no it doesn't mean that, an argoworkflows version that has api version of 0.24 (https://github.com/argoproj/argo-workflows/blob/v3.4.11/go.mod#L64) still runs fine on my 1.29 cluster. i'm sure 1.31 cluster will handle current argo-events just fine, upgrade your cluster!

**Comment by racterub** [comment]
In addition to k8s version compatibility, I think maybe we also need to include argo-workflow version in the matrix since argo-events should be coupled with argo-events.

It's clear that it can be verified through [code](https://github.com/argoproj/argo-events/blob/master/Dockerfile#L11), but I think it's can be added into both operation manual.

**Comment by shazib-summar** [comment]
I'd be more than happy to make and maintain the compatibility map but would require guidance on where to look for the compatible K8s versions.

From what @racterub says argo-events is coupled with argo-workflow so we should be looking for 

argo-event version -> corresponding argo workflow version -> argo workflows compatibility map

the argo workflows compatibility can be taken from: https://argo-workflows.readthedocs.io/en/stable/releases/#tested-versions

I'll wait for a +1 before starting any work on this
